contracts/src/v0.8/keystone/KeystoneFeedsPermissionHandler.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import {OwnerIsCreator} from "../shared/access/OwnerIsCreator.sol";
+
+/// @title Keystone Feeds Permission Handler
+/// @notice This contract is designed to manage and validate permissions for accessing specific reports within a decentralized system.
+/// @dev The contract uses mappings to keep track of report permissions associated with a unique report ID.
+abstract contract KeystoneFeedsPermissionHandler is OwnerIsCreator {
+  /// @notice Holds the details for permissions of a report
+  /// @dev Workflow names and report names are stored as bytes to optimize for gas efficiency.
+  struct Permission {
+    address forwarder;
+    bytes10 workflowName;
+    bytes2 reportName;
+    address workflowOwner;
+    bool isAllowed;
+  }
+
+  /// @notice Event emitted when report permissions are set
+  event ReportPermissionSet(bytes32 indexed reportId, Permission permission);
+
+  /// @notice Error to be thrown when an unauthorized access attempt is made
+  error Unauthorized(address forwarder, address workflowOwner, bytes10 workflowName, bytes2 reportName);
```

Comment:
suggestion: we should name this more specifically e.g. `ReportForwarderUnauthorized`, since there might be a clash with other auth-based functions

Reply:
This will be referenced as  `KeystoneFeedsPermissionHandler.Unauthorized` so there won't be a clash as such, but we can still rename it

Reply:
It's less about clashes, more about distinguishing the error.

When looking purely at the RPC logs, we would only see the error name and data, without the class / interface reference. `Unauthorized` could be confused with `AuthorizedCallers` methods, for example